packages/scheduler/src/forks/SchedulerDOM.js

```diff
@@ -547,15 +548,36 @@ const performWorkUntilDeadline = () => {
   needsPaint = false;
 };
 
-const channel = new MessageChannel();
-const port = channel.port2;
-channel.port1.onmessage = performWorkUntilDeadline;
+let schedulePerformWorkUntilDeadline;
+if (typeof setImmediate === 'function') {
```

Comment:
I think ideally we would create two separate bundles and resolve to the correct one at the packaging level. Maybe we can tackle this when we switch to ES Modules.